storage/coin_storage.go

```diff
@@ -444,3 +445,36 @@ func (c *CoinStorage) GetLargestCoin(
 
 	return bal, coinIdentifier, blockIdentifier, nil
 }
+
+// SetCoinsImported sets coins of a set of addresses by
+// getting their coins from the tip block, and populating the database.
+// This is used when importing prefunded addresses.
+func (c *CoinStorage) SetCoinsImported(
+	ctx context.Context,
+	helper BalanceStorageHelper,
+	accountBalances []*utils.AccountBalance,
+) error {
+	// Update balances in database
+	transaction := c.db.NewDatabaseTransaction(ctx, false)
+	defer transaction.Discard(ctx)
+
+	for _, accountBalance := range accountBalances {
+		for _, coin := range accountBalance.Coins {
+			encodedResult, err := encode(coin)
```

Comment:
This loop won't work...adding coins must also update the "wallet" held at each account

Reply:
You can add a test to see this by calling `GetCoins` on each account you add.